cpp/include/cugraph_c/graph.h

```diff
@@ -76,6 +76,46 @@ cugraph_error_code_t cugraph_sg_graph_create(
   cugraph_graph_t** graph,
   cugraph_error_t** error);
 
+/**
+ * @brief     Construct an SG graph from a CSR input
+ *
+ * @param [in]  handle      Handle for accessing resources
+ * @param [in]  properties  Properties of the constructed graph
+ * @param [in]  offsets     Device array containing the CSR offsets array
+ * @param [in]  indices     Device array containing the destination vertex ids
+ * @param [in]  weights     Device array containing the edge weights.  Note that an unweighted
+ *                          graph can be created by passing weights == NULL.
+ * @param [in]  edge_ids    Device array containing the edge ids for each edge.  Optional
+                            argument that can be NULL if edge ids are not used.
+ * @param [in]  edge_types  Device array containing the edge types for each edge.  Optional
```

Comment:
I wonder if it would be better if we started a naming convention that emphasizes these are arbitrary user-defined types, and not something to be confused with an `edge_t` used in so many places elsewhere.  I know most devs will see this and realize it can't really be an array of `edge_t` values, but it still might help with overall readability of our APIs IMO.

Maybe `edge_type_ids` instead of `edge_types`, with the "id" part to emphasize these are basically user-defined numerical enum values.

Related: this seems like it could have an impact on `edge_type_t` too?

Reply:
Updated in latest push